src/friendly-iframe-embed.js

```diff
@@ -229,24 +258,39 @@ export function installFriendlyIframeEmbed(
     iframe[FIE_EMBED_PROP] = embed;
 
     // Add extensions.
+    const extensionIds = spec.extensionIds || [];
     if (ampdoc && ampdocFieExperimentOn) {
       embed.installExtensionsInFie(
         extensions,
         ampdoc,
-        spec.extensionIds || [],
+        extensionIds,
         opt_preinstallCallback
       );
+
+      // Ready to be shown.
+      embed.startRender_();
+      return embed;
     } else {
-      embed.installExtensionsInChildWindow(
-        extensions,
-        childWin,
-        spec.extensionIds || [],
-        opt_preinstallCallback
-      );
+      return embed
+        .preInstallExtensionsInChildWindow(
+          extensions,
+          childWin,
+          extensionIds,
+          opt_preinstallCallback
+        )
+        .then(() => {
+          // Ready to be shown.
+          embed.startRender_();
```

Comment:
This is the thing to fix the loader race?

Reply:
Yes, this change ensures loader disappears a bit later, shortening the white screen time.